components/nimbus/src/tests/test_behavior.rs

```diff
@@ -1250,18 +1250,32 @@ mod event_store_tests {
             5.0,
             store.query(event_id, Interval::Days, 56, 0, EventQueryType::Sum)?
         );
-        assert_eq!(
-            5.0,
-            store.query(event_id, Interval::Weeks, 1, 0, EventQueryType::Sum)?
-        );
-        assert_eq!(
-            5.0,
-            store.query(event_id, Interval::Months, 1, 0, EventQueryType::Sum)?
-        );
-        assert_eq!(
-            5.0,
-            store.query(event_id, Interval::Years, 1, 0, EventQueryType::Sum)?
-        );
+        if same_year(now, then) {
+            assert_eq!(
+                5.0,
+                store.query(
+                    event_id,
+                    Interval::Weeks,
+                    1,
+                    usize::from(!same_week(now, then)), // 0 if the same week, 1 if not
+                    EventQueryType::Sum
```

Comment:
Just a nit but since these are SUM queries, we don't necessarily need to use the `same_<time>` methods and could just passively check two buckets instead of one. No problem leaving it like this though

Reply:
Added some comments and extra tests around using the appropriate granularity of interval.